### PR TITLE
feat(Boards)[OK-159] Implement user board tables and ui loading for board tables

### PR DESCRIPTION
**Pull request by adbista**
- Mock boards endpoint in BoardsFetcher.ts 
- Create more general Pagination for both Slides and Boards
- Adjust styles for badges

### CODE REVIEW

File: client/src/components/user-boards/TabButtons.tsx [hunk 1]
```diff
@@ -1,12 +1,13 @@
 "use client";
-import React, { useState } from "react";
+import React from "react";
 
-interface TabButtonProps {
+interface TabButton {
```
Comment (Strattoss) [inline client/src/components/user-boards/TabButtons.tsx]: Why have we decided to change this name from `TabButtonProps` to `TabButton`?

File: client/src/interfaces/responses/user-boards/board-response.ts [hunk 2]
```diff
@@ -0,0 +1,10 @@
+export interface BoardResponse {
+  id: string;
+  name: string;
+  modifiedAt: string;
+  createdAt: string;
+  size: number;
+  sharedWith: string[];
+  owner?: string;
+  yourPermission?: string;
```
Comment (Strattoss) [inline client/src/interfaces/responses/user-boards/board-response.ts]: I think we can safely rename it to `userPermission` or simply `permission`

File: client/src/components/ExtendedPagination.tsx
Comment (Strattoss) [inline client/src/components/ExtendedPagination.tsx:1]: I really hope that generalizing pagination like this doesn't backfire in the future 🤞
Especially if our lists use different systems (like `0, 1, 2, ...` vs `1, 2, 3, ...`)

**Review by Strattoss — APPROVED** [review]

File: client/src/interfaces/responses/user-boards/board-response.ts (hunk 2 above)
Comment (adbista) [reply client/src/interfaces/responses/user-boards/board-response.ts]: I need to rename that anyway to keep it consistent with the backend

File: client/src/components/user-boards/TabButtons.tsx (hunk 1 above)
Comment (adbista) [reply client/src/components/user-boards/TabButtons.tsx]: well... i see TabButtonProps

File: client/src/components/ExtendedPagination.tsx
Comment (adbista) [reply client/src/components/ExtendedPagination.tsx:1]: image

As you can see, the numbers are only incremented by 1 when pagination is displayed.

File: client/src/components/user-boards/TabButtons.tsx (hunk 1 above)
Comment (adbista) [reply client/src/components/user-boards/TabButtons.tsx]: Oh yeah, this comment is from 3 days ago.

**Pull request merged**